src/components/Toolbar.js

```diff
@@ -8,57 +8,119 @@ import {
   Body,
   Icon,
 } from 'native-base';
+import { View, StyleSheet } from 'react-native';
 
-export default class Toolbar extends React.Component {
+import { SearchBar } from './SearchBar';
 
-  createLeftButton(){
-    return (
-      <Left>
-        <Button transparent onPress={ () => {
-          switch (this.props.leftButtonIcon){
-            case 'menu' :
-              this.props.navigation.navigate("DrawerOpen");
-              break;
-            case 'arrow-back' :
-              this.props.navigation.goBack();
-              break;
-          }
-        }}>
-          <Icon name={this.props.leftButtonIcon} />
-        </Button>
-      </Left>
-    );
+export default class Toolbar extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFilterActive: false,
+    };
   }
 
-  createRightButton(){
+  createButton(button) {
     return (
-      <Right>
-         <Button transparent>
-          <Icon name={this.props.rightButtonIcon} />
-         </Button>
-      </Right>
-    );
+      <Button 
+        transparent 
+        onPress={() => {
+          if (button.icon === 'search') {
+            this.setState({
+              isFilterActive: true,
+            })
+          } else {
+            button.action();
+          }      
+        }}
+      >
```

Comment:
i don't think that a function would be necessary because it is only an if sentence that executes the button action if it is not creating the search button but in the search button case his action must be different from the others.